HW_4/download_img.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+from urllib.parse import urlparse
+import time
+import requests
+import threading
+from multiprocessing import Process
+import asyncio
+import aiohttp
+
+
+def_urls_list = [
+    'https://free-png.ru/wp-content/uploads/2021/07/free-png.ru-30.png',
+    'https://dayonline.ru/public/article/images/2c7d53a242a05059a14677e8c0dccd788f7efa70.jpg',
+    'https://w.forfun.com/fetch/4e/4e73e36e635ef13ae01d945d96369f8b.jpeg'
+]
+
+urls_list = []
+
+class WebImage():
+    def __init__(self, url, file_name):
+        self.url_str = url
+        self.file_name_str = file_name
+
+    @property
+    def url(self):
+        return self.url_str
+
+    @property
+    def file_name(self):
+        return self.file_name_str
+
+
+def time_complete(func):
+    def wrapper():
+        time_start = time.time()
+        func()
+        time_end = time.time()
+        print(f'Метод {func.__name__}: Изображения загружены за {time_end - time_start:.2f} секунд')
+
+    return wrapper
+
+
+def url_parse(strings):  # -> list[WebImage]:
+    result = list()
+    for url_str in strings:
+        url = urlparse(url_str)
+        file_name = url.path.split('/')[-1]
```

Comment:
Можно было и без функции urlparse обойтись. Она все равно особо не помогает, только кода кода
```suggestion
        file_name = url_str.split('/')[-1]
```
или так
```suggestion
        file_name = url_str.rsplit('/', 1)[-1]
```
или
```suggestion
        file_name = os.path.basename(url_str)
```